contributing/dev_env_setup.sh

```diff
@@ -0,0 +1,176 @@
+    ./rtf platform setup debian9_dev
+    # will take a while...
+    netstat -laputn | grep 15432 # check purpose only, should print : `tcp        0      0 0.0.0.0:15432           0.0.0.0:*               LISTEN`
+    cd $rudder_dir/
+
+####################################
+# NCF API & TECHNIQUE EDITOR SETUP #
+####################################
+
+    # setup ncf api
+    wget https://repository.rudder.io/build-dependencies/virtualenv/virtualenv-16.5.0.tar.gz
+    tar xzvf virtualenv-16.5.0.tar.gz
+    rm -rf virtualenv-16.5.0.tar.gz
+    mv virtualenv-16.5.0/* $rudder_dir/ncf/api/
+    cd $rudder_dir/ncf/api/
+    python virtualenv.py flask
+    flask/bin/pip install -r requirements.txt
+    # setup technique editor
+    # depending on your distrib, folder might be: `/etc/apache2/conf.d/`
+    sudo cp $rudder_dir/ncf/api/dev_env/ncf-builder.conf /etc/apache2/conf-enabled/
+    sudo sed -i "s/8042/8080/g" /etc/apache2/conf-enabled/ncf-builder.conf
+    sudo sed -i "s/\/path\/to\/ncf\/builder/\/home\/$username\/rudder\/ncf\/builder/g" /etc/apache2/conf-enabled/ncf-builder.conf
+    chmod 755 $rudder_dir/ncf/builder -R
+    sudo a2enmod rewrite
+    sudo a2enmod proxy
+    sudo a2enmod proxy_http
+    sudo service apache2 restart
+
+###########################
+# PRE SETUP FOR SOFTWARES #
+###########################
+
+    mkdir -p /var/rudder/inventories/incoming /var/rudder/share /var/rudder/inventories/accepted-nodes-updates /var/rudder/inventories/received /var/rudder/inventories/failed /var/log/rudder/core /var/log/rudder/compliance/ /var/rudder/run/
+    touch /var/log/rudder/core/rudder-webapp.log /var/log/rudder/compliance/non-compliant-reports.log /var/rudder/run/api-token
+    # since a lot has been installed as root apps launched as $username could need permissions to access its home space
+    sudo chown $username -R /var/rudder/
+    sudo chown $username -R /var/log/rudder
+    sudo chown $username -R /home/gpoblon/
+
+    cp -R $rudder_dir/rudder-techniques/techniques /var/rudder/configuration-repository/
+    # setup maven dependencies in intellij
+    cp $rudder_dir/rudder/contributing/settings.xml /home/$username/.m2/settings.xml
+    sed -i "s/\[PATH TO \.m2 DIRECTORY\]/\/home\/$username\//g" /home/$username/.m2/settings.xml
+    #takes a while
+    cd $rudder_dir/rudder/webapp/sources && mvn clean install
+
+##########################
+# VAGRANT VM SLAPD SETUP #
+##########################
+
+    # connect to newly created VM
+    cd $rudder_dir/rudder-tests/
+    vagrant ssh debian9_dev_server -c "echo -e '[Service]\nExecStart=\nExecStart=/opt/rudder/libexec/slapd -n rudder-slapd -u rudder-slapd -f /opt/rudder/etc/openldap/slapd.conf -h \"ldap://0.0.0.0:389/\"' | sudo tee /etc/systemd/system/rudder-slapd.service.d/override.conf"
```

Comment:
So Strange I thought a i made a PR about that somewhere ...

Reply:
Well maybe the documentation was not up to date but when I set up the vagrant VM this file already had the right content.

So it is a "just in case" command.

Maybe we should simply remove it from the webapp, if there is no chance override.conf has to be rewritten

Reply:
It was before your changes @VinceMacBuche, we doesn't need that part anymore i think

Reply:
you can remove it then

Reply:
ok ! So this can disappear too ? : sudo systemctl daemon-reload && sudo service rudder-slapd restart

Reply:
Yes :)